modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/create_folder_command.rb

```diff
@@ -32,52 +32,96 @@ module Storages::Peripherals::StorageInteraction::Nextcloud
   class CreateFolderCommand
     using Storages::Peripherals::ServiceResultRefinements
 
-    def initialize(storage)
-      @uri = storage.uri
-      @username = storage.username
-      @password = storage.password
+    Auth = ::Storages::Peripherals::StorageInteraction::Authentication
+
+    def self.call(storage:, auth_strategy:, folder_name:, parent_location:)
+      new(storage).call(auth_strategy:, folder_name:, parent_location:)
     end
 
-    def self.call(storage:, folder_path:)
-      new(storage).call(folder_path:)
+    def initialize(storage)
+      @storage = storage
     end
 
-    # rubocop:disable Metrics/AbcSize
-    def call(folder_path:)
-      response = OpenProject
-                   .httpx
-                   .basic_auth(@username, @password)
-                   .mkcol(
-                     Util.join_uri_path(
-                       @uri,
-                       "remote.php/dav/files",
-                       CGI.escapeURIComponent(@username),
-                       Util.escape_path(folder_path)
-                     )
-                   )
+    def call(auth_strategy:, folder_name:, parent_location:)
+      origin_user_id = Util.origin_user_id(caller: self.class, storage: @storage, auth_strategy:)
+      if origin_user_id.failure?
+        return origin_user_id
+      end
+
+      folder_path = Util.join_uri_path(parent_location, folder_name)
+      base_url = Util.join_uri_path(@storage.uri, "remote.php/dav/files", CGI.escapeURIComponent(origin_user_id.result))
+      path_prefix = URI.parse(base_url).path
+      request_url = Util.join_uri_path(base_url, Util.escape_path(folder_path))
+
+      Auth[auth_strategy].call(storage: @storage) do |http|
+        result = handle_response(http.mkcol(request_url))
+        return result if result.failure?
+
+        handle_response(http.propfind(request_url, requested_properties)).map(&storage_file(path_prefix))
```

Comment:
🟡 I may be alone on this, but I find link chains hard to parse/understand, specially with the weird `map(&method(param))` in the end.

`map` takes a block, why not use one so you can remove the inner lambda from the method?

```ruby
service_result.map { |result| storage_file(result, path_prefix) }
```

looks clearer and conveys more intent as you see that it is working over the `result` and needs one extra parameter.